plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/wst/xml/ui/tabletree/XMLMultiPageEditor.java

```diff
@@ -0,0 +1,108 @@
+package org.jboss.reddeer.eclipse.wst.xml.ui.tabletree;
+
+import org.eclipse.ui.IEditorPart;
+import org.eclipse.ui.part.MultiPageEditorPart;
+import org.eclipse.ui.texteditor.ITextEditor;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeMatcher;
+import org.jboss.reddeer.core.matcher.WithTextMatcher;
+import org.jboss.reddeer.core.util.Display;
+import org.jboss.reddeer.core.util.ResultRunnable;
+import org.jboss.reddeer.eclipse.exception.EclipseLayerException;
+import org.jboss.reddeer.eclipse.ui.part.MultiPageEditor;
+
+/**
+ * Represents an XML editor with Design and Source pages.
+ * 
+ * @author Lucia Jelinkova
+ *
+ */
+public class XMLMultiPageEditor extends MultiPageEditor {
+
+	/**
+	 * Find XMLMultiPageEditorPart with the given title
+	 * @param title
+	 */
+	public XMLMultiPageEditor(String title) {
+		this(new WithTextMatcher(title));
+	}
+	
+	/**
+	 * Find XMLMultiPageEditorPart with the given title matcher
+	 * @param title
+	 */
+	@SuppressWarnings("unchecked")
+	public XMLMultiPageEditor(Matcher<String> titleMatcher) {
+		super(titleMatcher, MultiPageEditorPart.class, new XMLMultiPageEditorPartMatcher());
+	}
+	
+	/**
+	 * Select the design page tab
+	 */
+	public void selectDesignPage(){
+		super.selectPage("Design");
+	}
+	
+	/**
+	 * Select the source page tab
+	 */
+	public void selectSourcePage(){
+		super.selectPage("Source");
+	}
+	
+	/**
+	 * Return object for working with design page
+	 * @return page that enables to work with XML in tree format
+	 */
+	public XMLDesignPage getDesignPage(){
+		selectDesignPage();
+		return new XMLDesignPage();
+	}
+	
+	/**
+	 * Return object for working with source page
+	 * @return page that enables to work with XML in text format
+	 */
+	public XMLSourcePage getSourcePage(){
+		selectSourcePage();
+		Object o = getSelectedPage();
+		if (o instanceof ITextEditor){
+			return new XMLSourcePage((ITextEditor) o);
+		} 
+		throw new EclipseLayerException("Expected " + ITextEditor.class + 
+				" but was " + o.getClass());
+	}
+	
+	
+	private Object getSelectedPage(){
+		return Display.syncExec(new ResultRunnable<Object>() {
+
+			@Override
+			public Object run() {
+				return ((MultiPageEditorPart) getEditorPart()).getSelectedPage();
+			}
+		});
+	}
```

Comment:
This will not work for subclasses I guess. Class of editor will be different then FULL_CLASS_NAME so this matcher will always return false